### DISCUSSION TITLE
Operations with calc on CSS properties

### DISCUSSION THREAD

**Discussion by michel-leanon** [chakra-ui/chakra-ui#5216] [General] [answered]
Hey!

Can I use the "calc" function to calculate sizes on CSS?

CSS Example:
`width: calc(100vh - 150px);`

Is it possible on ChakraUI ?

**Comment by malashevskyi** [comment, marked as answer]
Why not? I recently used `calc` to change modal size
 [example](https://codesandbox.io/s/chakra-ui-collapse-modal-sidebar-with-width-gmyer?file=/src/App.js)
```javascript
<ModalContent
  maxW="calc(100vw - 50px)"
  minH="calc(100vh - 25px)"
  my="auto"
>
```

**Reply by Shaad-Industries** [reply]
Great it's working.👍

**Comment by sheryarshirazi** [comment]
@michel-leanon
how can measure units in calc function? like 16

```
<Box as="header" h={16} />
<Box
  minH="calc(100vh - 4rem)"
>
```

**Reply by tannerbaum** [reply]
As far as I know, you have to use their css variable names. So for example --chakra-sizes-16 (or whatever it is called)

**Reply by sheryarshirazi** [reply]
thanks @tannerbaum , it works.

**Comment by jruizsilva** [comment]
If you want to use calc with percentages you can try
```
<Box
  width={calc('100%').subtract('500px').toString()}
>
```
This also works
`
```
<Box
  width={`${calc(100% -500).toString()}px`}
>
```
I hope it works for you
